internal/provider/connection/sdk.go

```diff
@@ -1,209 +1,594 @@
+		}
+		return diags
+	}
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		diags.AddError("Error updating connection", err.Error())
+		return diags
+	}
+
+	var connection map[string]interface{}
+	err = json.Unmarshal(body, &connection)
+	if err != nil {
+		diags.AddError("Error updating connection", err.Error())
+		return diags
+	}
+
+	return m.Refresh(connection)
+}
+
+func (m *connectionResourceModel) Delete(ctx context.Context, client *sdkclient.Client) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	response, err := client.RawClient.SendRequest(ctx, "DELETE", fmt.Sprintf("/%s/connections/%s", apiVersion, m.ID.ValueString()), &sdkclient.RequestOptions{})
+	if err != nil {
+		diags.AddError("Error deleting connection", err.Error())
+		return diags
+	}
+
+	if response.StatusCode > 299 {
+		if body, err := io.ReadAll(response.Body); err == nil {
+			diags.AddError("Error deleting connection", string(body))
+		} else {
+			diags.AddError("Error deleting connection", "Status code: "+strconv.Itoa(response.StatusCode))
+		}
+		return diags
 	}
+
+	return nil
 }
 
-func (m *connectionResourceModel) ToCreatePayload() *hookdeck.ConnectionCreateRequest {
-	return &hookdeck.ConnectionCreateRequest{
-		Name:          hookdeck.OptionalOrNull(m.Name.ValueStringPointer()),
-		Description:   hookdeck.OptionalOrNull(m.Description.ValueStringPointer()),
-		DestinationId: hookdeck.OptionalOrNull(m.DestinationID.ValueStringPointer()),
-		Rules:         hookdeck.OptionalOrNull(m.getRules()),
-		SourceId:      hookdeck.OptionalOrNull(m.SourceID.ValueStringPointer()),
+// toCreatePayload converts the model to API create payload.
+func (m *connectionResourceModel) toCreatePayload(ctx context.Context) (map[string]interface{}, error) {
+	payload := map[string]interface{}{}
+
+	if !m.Name.IsNull() && !m.Name.IsUnknown() {
+		payload["name"] = m.Name.ValueString()
+	}
+	if !m.Description.IsNull() && !m.Description.IsUnknown() {
+		payload["description"] = m.Description.ValueString()
+	}
+	if !m.DestinationID.IsNull() && !m.DestinationID.IsUnknown() {
+		payload["destination_id"] = m.DestinationID.ValueString()
 	}
+	if !m.SourceID.IsNull() && !m.SourceID.IsUnknown() {
+		payload["source_id"] = m.SourceID.ValueString()
+	}
+
+	// Convert rules - only include if not nil
+	if m.Rules != nil {
+		apiRules, err := rulesToAPI(ctx, m.Rules)
+		if err != nil {
+			return nil, err
+		}
+		payload["rules"] = apiRules
+	}
+
+	return payload, nil
 }
 
-func (m *connectionResourceModel) ToUpdatePayload() *hookdeck.ConnectionUpdateRequest {
-	return &hookdeck.ConnectionUpdateRequest{
-		Name:        hookdeck.OptionalOrNull(m.Name.ValueStringPointer()),
-		Description: hookdeck.OptionalOrNull(m.Description.ValueStringPointer()),
-		Rules:       hookdeck.OptionalOrNull(m.getRules()),
+// toUpdatePayload converts the model to API update payload.
+func (m *connectionResourceModel) toUpdatePayload(ctx context.Context) (map[string]interface{}, error) {
```

Comment:
comment: I was unsure whether this was going to be used with a `PUT` request to `/connections` or `/connection/:id`. I found it was the latter, which made sense based on the properties being updated.

Potentially no action required from this comment, just something to think on.